### PR TITLE
Magicarp named after their bolts

### PR DESCRIPTION
**Pull request by coiax**
That's a bolt of death magicarp
50% magic, 50% carp, 100% horrible.

🆑 coiax
add: Magicarp are now named after the type of magical bolt that they
shoot at people.
/🆑

Minor buff to Xenobio, as you can now determine a carp's bolt type
without waking it up.

Chaos magicarp are unaffected, and are still called chaos magicarp.

### CODE REVIEW

File: code/modules/events/wizard/magicarp.dm [hunk 1]
```diff
@@ -39,18 +39,23 @@
 	var/allowed_projectile_types = list(/obj/item/projectile/magic/change, /obj/item/projectile/magic/animate, /obj/item/projectile/magic/resurrection,
 	/obj/item/projectile/magic/death, /obj/item/projectile/magic/teleport, /obj/item/projectile/magic/door, /obj/item/projectile/magic/aoe/fireball,
 	/obj/item/projectile/magic/spellblade, /obj/item/projectile/magic/arcane_barrage)
+	var/chaos = FALSE
 	
 /mob/living/simple_animal/hostile/carp/ranged/Initialize()
 	projectiletype = pick(allowed_projectile_types)
+	if(!chaos && initial(name) == name)
+		name = "[initial(projectiletype.name)] [name]"
```
Comment (Cyberboss) [inline code/modules/events/wizard/magicarp.dm]: Likewise, refactor this into a proc which is empty on `/chaos` and remove the var

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 2]
```diff
@@ -14,7 +14,7 @@
 	var/dodge_prob = 30
 	var/sidestep_per_cycle = 1 //How many sidesteps per npcpool cycle when in melee
 
-	var/projectiletype	//set ONLY it and NULLIFY casingtype var, if we have ONLY projectile
+	var/obj/item/projectile/projectiletype	//set ONLY it and NULLIFY casingtype var, if we have ONLY projectile
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]:
```suggestion
	var/projectiletype	//set ONLY it and NULLIFY casingtype var, if we have ONLY projectile
```

We don't type path variables

File: code/modules/events/wizard/magicarp.dm [hunk 3]
```diff
@@ -39,18 +39,23 @@
 	var/allowed_projectile_types = list(/obj/item/projectile/magic/change, /obj/item/projectile/magic/animate, /obj/item/projectile/magic/resurrection,
 	/obj/item/projectile/magic/death, /obj/item/projectile/magic/teleport, /obj/item/projectile/magic/door, /obj/item/projectile/magic/aoe/fireball,
 	/obj/item/projectile/magic/spellblade, /obj/item/projectile/magic/arcane_barrage)
+	var/chaos = FALSE
```
Comment (Cyberboss) [inline code/modules/events/wizard/magicarp.dm]:
```suggestion
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Wilchenx** [comment]
>taking experimenting out of the job thats supposed to be around experimenting

whats next? giving everyone ability to see what chems are inside pills/containers without science googles/beer glasses?

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Also, the name shouldn't be set without waking it up

**Comment by coiax** [comment]
What if we just colored them according to the bolts they shot?

**Comment by anconfuzedrock** [comment]
I think if one scientist wants to make a carp that can outdo the entirety of medbay, making a whole department useless, just spending a while on xenobio shouldn't be enough. he's going to at least have to see face-first if that carp shoots bolts of death or of healing.

**Comment by coiax** [comment]
huh, okay

**Pull request closed without merging**